docs/clients/dotnet/21.2/migration-to-gRPC.md

```diff
@@ -0,0 +1,599 @@
+- removing TCP client package reference as the last step.
+
+You may also consider wrapping common logic into extension methods or repository classes. Adding a temporary wrapping layer lets us centralise and isolate the changes we'll be performing during the migration. Thanks to that, you can replace the inner implementations, keeping usages the same. Read more in Martin Fowler's article [An example of preparatory refactoring](https://martinfowler.com/articles/preparatory-refactoring-example.html).
+
+Sample wrapper for the TCP client may look, e.g.:
+
+```csharp
+public class EventStore
+{
+    readonly IEventStoreConnection _connection;
+
+    public EventStore(IEventStoreConnection connection)
+        => _connection = connection;
+
+    public Task AppendEvents(
+        string streamName,
+        long version,
+        params object[] events
+    )
+    {
+        var preparedEvents = events
+            .Select(ToEventData)
+            .ToArray();
+
+        return _connection.AppendToStreamAsync(
+            streamName,
+            version,
+            preparedEvents
+        );
+
+        static EventData ToEventData(object @event) =>
+            new EventData(
+                Guid.NewGuid(),
+                TypeMapper.GetTypeName(@event.GetType()),
+                true,
+                Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(data))
+            );
+    }
+
+    public Task AppendEvents(
+        string streamName,
+        params object[] events
+    )
+        => AppendEvents(streamName, ExpectedVersion.Any, events);
+
+    public async Task<IEnumerable<object>> LoadEvents(string stream)
+    {
+        const int pageSize = 4096;
+
+        var start  = 0;
+        var events = new List<object>();
+
+        do
+        {
+            var page = await _connection.ReadStreamEventsForwardAsync(
+                stream, start, pageSize, true
+            );
+
+            if (page.Status == SliceReadStatus.StreamNotFound)
+                throw new ArgumentOutOfRangeException(
+                    nameof(stream), $"Stream '{stream}' was not found"
+                );
+
+            events.AddRange(
+                page.Events.Select(Deserialize)
+            );
+            if (page.IsEndOfStream) break;
+
+            start += pageSize;
+        } while (true);
+
+        return events;
+
+        static object Deserialize(this ResolvedEvent resolvedEvent)
+        {
+            var dataType = TypeMapper.GetType(resolvedEvent.Event.EventType);
+            var jsonData = Encoding.UTF8.GetString(resolvedEvent.Event.Data);
+            var data = JsonConvert.DeserializeObject(jsonData, dataType);
+            return data;
+        }
+    }
+
+    public async Task<bool> StreamExists(string stream)
+    {
+        var result = await _connection.ReadEventAsync(stream, 1, false);
+        return result.Status != EventReadStatus.NoStream;
+    }
+}
+```
+
+## Differences in Connection Management
```

Comment:
We don't use capital casing for titles, I keep fixing it everywhere....

Reply:
Fixed in https://github.com/EventStore/documentation/pull/486/commits/ffc35a21c4c86f91e92896e4d467dece8d636351.